modules/tensor_mechanics/src/dampers/ReferenceElementJacobianDamper.C

```diff
@@ -0,0 +1,115 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReferenceElementJacobianDamper.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "Assembly.h"
+#include "SystemBase.h"
+
+#include "libmesh/quadrature.h" // _qrule
+
+registerMooseObject("TensorMechanicsApp", ReferenceElementJacobianDamper);
+
+InputParameters
+ReferenceElementJacobianDamper::validParams()
+{
+  InputParameters params = GeneralDamper::validParams();
+  params.addClassDescription("Damper that limits the change in element Jacobians");
+  params.addRequiredCoupledVar("displacements", "The nonlinear displacement variables");
+  params.addParam<Real>(
+      "max_increment", 0.1, "The maximum permissible incremental Jacobian per Newton iteration");
+  return params;
+}
+
+ReferenceElementJacobianDamper::ReferenceElementJacobianDamper(const InputParameters & parameters)
+  : GeneralDamper(parameters),
+    Coupleable(this, /*nodal=*/true),
+    _max_jacobian_diff(getParam<Real>("max_increment")),
+    _tid(getParam<THREAD_ID>("_tid")),
+    _mesh(_subproblem.mesh()),
+    _assembly(_subproblem.assembly(_tid)),
+    _qrule(_assembly.qRule()),
+    _ndisp(coupledComponents("displacements")),
+    _disp_num(coupledIndices("displacements"))
+{
+  _grad_phi.resize(_ndisp);
```

Comment:
```suggestion
  if (getParam<bool>("use_displaced_mesh"))
    paramError("use_displaced_mesh",
               "ReferenceElementJacobianDamper needs to act on an undisplaced mesh.");
  _grad_phi.resize(_ndisp);
```

Reply:
Ummm, no. `GeneralDamper` doesn't have the `use_displaced_mesh` option. Users won't be able to set that. If they set it they get an unused param warning.

Reply:
Oh, yeah, makes sense!